src/settings.rs

```diff
@@ -1,18 +1,25 @@
 //! The global variables of this project.
 //!
 //! This module specifies the magic numbers of this project once and centrally
-//! in order to avoid redundancy.
+//! in order to avoid redundancy and to enhance the maintainability.
+//!
+//! The constants are grouped by the following categories:
+//!
+//! * `application`: application related information, and
+//! * `version`: version information
 
-/// This application's name.
-#[allow(dead_code)]
-pub const APP_NAME: &str = "commit-analyzer";
+/// Application related settings.
+pub mod application {
+    /// This application's name.
+    pub const NAME: &str = "commit-analyzer";
 
-/// This application's tool tip.
-///
-/// In order to show a brief usage synopsis to the user, this global variable
-/// sets the information about the required and optional command line arguments.
-#[allow(dead_code)]
-pub const APP_TOOL_TIP: &str = "<FILE> [OPTIONS]";
+    /// This application's tool tip.
+    ///
+    /// In order to show a brief usage synopsis to the user, this global
+    /// variable sets the information about the required and optional command
+    /// line arguments.
+    pub const TOOL_TIP: &str = "<FILE> [OPTIONS]";
+}
```

Comment:
Yes, this is why I moved all modules to `src/lib.rs`.

And you will definitely need those module definitions for unit testing, see Chapter 11 of the Rust book (I always refer to the printed version which covers Rust 2018).

Reply:
No, i mean using the 
```rust
mod name {
/* content */
}```
I personally prefer the
```rust
mod name;```
syntax, since code is less indented that way. But to not cluster everything with files one should to this only if a module would be big enough.. 😅 

Regarding tests, you can obviously write them also like this, even though the book does not show that:
```rust
#[cfg(test)]
mod test;
```
but I don't know if that would be proper style. I've only used tests in one rust project yet..

Reply:
Basically just pulling everything out of the modules would solve this issue, right? :wrench:

Reply:
Yes 👍

Reply:
:wrench: :heavy_check_mark: